src/uu/split/src/split.rs

```diff
@@ -288,6 +311,142 @@ fn num_prefix(i: usize, width: usize) -> String {
     c
 }
 
+#[cfg(unix)]
+/// A writer that writes to a shell_process' stdin
+///
+/// We use a shell process (not directy calling a sub-process) so we can forward the name of the
+/// corresponding output file (xaa, xab, xac… ). This is the way it was implemented in GNU split.
+struct FilterWriter {
+    /// Running shell process
+    shell_process: Child,
+}
+
+#[cfg(unix)]
+impl Write for FilterWriter {
+    fn write(&mut self, buf: &[u8]) -> Result<usize> {
+        self.shell_process
+            .stdin
+            .as_mut()
+            .expect("failed to get shell stdin")
+            .write(buf)
+    }
+    fn flush(&mut self) -> Result<()> {
+        self.shell_process
+            .stdin
+            .as_mut()
+            .expect("failed to get shell stdin")
+            .flush()
+    }
+}
+
+/// Have an environment variable set at a value during this lifetime
+#[cfg(unix)]
+struct WithEnvVarSet {
+    /// Env var key
+    _previous_var_key: String,
+    /// Previous value set to this key
+    _previous_var_value: std::result::Result<String, env::VarError>,
+}
+#[cfg(unix)]
+impl WithEnvVarSet {
+    /// Save previous value assigned to key, set key=value
+    fn new(key: &str, value: &str) -> WithEnvVarSet {
+        let previous_env_value = env::var(key);
+        env::set_var(key, value);
+        WithEnvVarSet {
+            _previous_var_key: String::from(key),
+            _previous_var_value: previous_env_value,
+        }
+    }
+}
+#[cfg(unix)]
```

Comment:
this is a lot of "cfg(unix)"
what about moving that into a specific platform file?
like
https://github.com/uutils/coreutils/tree/master/src/uu/tail/src/platform/

Reply:
sure, no problem